client/my-sites/stats/controller.jsx

```diff
@@ -378,5 +378,30 @@ module.exports = {
 				context.store
 			);
 		}
+	},
+
+	activity_log: function( context ) {
+		let siteId = context.params.site_id;
+		const site = getSite( context.store.getState(), siteId );
+		siteId = site ? ( site.ID || 0 ) : 0;
```

Comment:
Actually, the logic here does not seem to be working. I wasn't able to reach this route without hitting the redirect (`( siteId ==0 || ! isJetpack` ). I always got redirected to `/stats`. This is reproducible consistently by cleaning the browser's storage and trying to visit `/stats/activity/:site`.